ccl/changefeedccl: TestChangefeedMultiTable failed under stress

SHA: https://github.com/cockroachdb/cockroach/commits/956889269f956a74bc9f075663c06d0f0c7de830

Parameters:
```
TAGS=
GOFLAGS=-race
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=840408&tab=buildLog

```
=== RUN   TestChangefeedMultiTable
W180816 05:21:29.717560 1420 server/status/runtime.go:294  [n?] Could not parse build timestamp: parsing time "" as "2006/01/02 15:04:05": cannot parse "" as "2006"
I180816 05:21:29.840147 1420 server/server.go:827  [n?] monitoring forward clock jumps based on server.clock.forward_jump_check_enabled
I180816 05:21:29.840714 1420 base/addr_validation.go:260  [n?] server certificate addresses: IP=127.0.0.1,::1; DNS=localhost,*.local; CN=node
I180816 05:21:29.840797 1420 base/addr_validation.go:300  [n?] web UI certificate addresses: IP=127.0.0.1,::1; DNS=localhost,*.local; CN=node
I180816 05:21:29.854222 1420 server/config.go:496  [n?] 1 storage engine initialized
I180816 05:21:29.854316 1420 server/config.go:499  [n?] RocksDB cache size: 128 MiB
I180816 05:21:29.854369 1420 server/config.go:499  [n?] store 0: in-memory, size 0 B
I180816 05:21:29.956199 1420 server/node.go:373  [n?] **** cluster c04613dd-addc-439f-8f56-73e6767bb7e4 has been created
I180816 05:21:29.956303 1420 server/server.go:1398  [n?] **** add additional nodes by specifying --join=127.0.0.1:35017
I180816 05:21:29.957115 1420 gossip/gossip.go:383  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:35017" > attrs:<> locality:<> ServerVersion:<major_val:2 minor_val:0 patch:0 unstable:11 > build_tag:"v2.1.0-alpha.20180702-1631-g9568892" started_at:1534396889956809155 
I180816 05:21:29.985759 1420 storage/store.go:1506  [n1,s1] [n1,s1]: failed initial metrics computation: [n1,s1]: system config not yet available
I180816 05:21:29.986764 1420 server/node.go:476  [n1] initialized store [n1,s1]: disk (capacity=512 MiB, available=512 MiB, used=0 B, logicalBytes=6.9 KiB), ranges=1, leases=1, queries=0.00, writes=0.00, bytesPerReplica={p10=7103.00 p25=7103.00 p50=7103.00 p75=7103.00 p90=7103.00 pMax=7103.00}, writesPerReplica={p10=0.00 p25=0.00 p50=0.00 p75=0.00 p90=0.00 pMax=0.00}
I180816 05:21:29.987034 1420 storage/stores.go:222  [n1] read 0 node addresses from persistent storage
I180816 05:21:29.987427 1420 server/node.go:697  [n1] connecting to gossip network to verify cluster ID...
I180816 05:21:29.987569 1420 server/node.go:722  [n1] node connected via gossip and verified as part of cluster "c04613dd-addc-439f-8f56-73e6767bb7e4"
I180816 05:21:30.006085 1420 server/node.go:546  [n1] node=1: started with [<no-attributes>=<in-mem>] engine(s) and attributes []
I180816 05:21:30.013037 1469 storage/replica_command.go:282  [split,n1,s1,r1/1:/M{in-ax}] initiating a split of this range at key /System/"" [r2]
I180816 05:21:30.056837 1420 server/status/recorder.go:652  [n1] available memory from cgroups (8.0 EiB) exceeds system memory 16 GiB, using system memory
I180816 05:21:30.056941 1420 server/server.go:1833  [n1] Could not start heap profiler worker due to: directory to store profiles could not be determined
I180816 05:21:30.058724 1420 server/server.go:1535  [n1] starting https server at 127.0.0.1:46815 (use: 127.0.0.1:46815)
I180816 05:21:30.058811 1420 server/server.go:1537  [n1] starting grpc/postgres server at 127.0.0.1:35017
I180816 05:21:30.058859 1420 server/server.go:1538  [n1] advertising CockroachDB node at 127.0.0.1:35017
W180816 05:21:30.059055 1420 jobs/registry.go:299  [n1] unable to get node liveness: node not in the liveness table
I180816 05:21:30.083971 1658 server/status/recorder.go:652  [n1,summaries] available memory from cgroups (8.0 EiB) exceeds system memory 16 GiB, using system memory
I180816 05:21:30.169224 1322 storage/replica_command.go:282  [split,n1,s1,r2/1:/{System/-Max}] initiating a split of this range at key /System/NodeLiveness [r3]
W180816 05:21:30.313007 1671 storage/intent_resolver.go:668  [n1,s1] failed to push during intent resolution: failed to push "sql txn" id=0e2ebaa8 key=/Table/SystemConfigSpan/Start rw=true pri=0.00238237 iso=SERIALIZABLE stat=PENDING epo=0 ts=1534396890.163577609,0 orig=1534396890.163577609,0 max=1534396890.163577609,0 wto=false rop=false seq=6
I180816 05:21:30.314638 1675 storage/replica_command.go:282  [split,n1,s1,r3/1:/{System/NodeL…-Max}] initiating a split of this range at key /System/NodeLivenessMax [r4]
I180816 05:21:30.385846 1319 sql/event_log.go:126  [n1,intExec=optInToDiagnosticsStatReporting] Event: "set_cluster_setting", target: 0, info: {SettingName:diagnostics.reporting.enabled Value:true User:root}
I180816 05:21:30.831808 1377 storage/replica_command.go:282  [split,n1,s1,r4/1:/{System/NodeL…-Max}] initiating a split of this range at key /System/tsd [r5]
I180816 05:21:30.928290 1407 storage/replica_command.go:282  [split,n1,s1,r5/1:/{System/tsd-Max}] initiating a split of this range at key /System/"tse" [r6]
I180816 05:21:31.055616 1329 storage/replica_command.go:282  [split,n1,s1,r6/1:/{System/tse-Max}] initiating a split of this range at key /Table/SystemConfigSpan/Start [r7]
I180816 05:21:31.102256 1705 sql/event_log.go:126  [n1,intExec=set-setting] Event: "set_cluster_setting", target: 0, info: {SettingName:version Value:$1 User:root}
I180816 05:21:31.170799 1325 rpc/nodedialer/nodedialer.go:92  [consistencyChecker,n1,s1,r1/1:/{Min-System/}] connection to n1 established
I180816 05:21:31.232461 1764 storage/replica_command.go:282  [split,n1,s1,r7/1:/{Table/System…-Max}] initiating a split of this range at key /Table/11 [r8]
I180816 05:21:31.395356 1722 storage/replica_command.go:282  [split,n1,s1,r8/1:/{Table/11-Max}] initiating a split of this range at key /Table/12 [r9]
I180816 05:21:31.494892 1762 sql/event_log.go:126  [n1,intExec=disableNetTrace] Event: "set_cluster_setting", target: 0, info: {SettingName:trace.debug.enable Value:false User:root}
I180816 05:21:31.574900 1780 storage/replica_command.go:282  [split,n1,s1,r9/1:/{Table/12-Max}] initiating a split of this range at key /Table/13 [r10]
I180816 05:21:31.686100 1830 storage/replica_command.go:282  [split,n1,s1,r10/1:/{Table/13-Max}] initiating a split of this range at key /Table/14 [r11]
I180816 05:21:31.768077 1818 storage/replica_command.go:282  [split,n1,s1,r11/1:/{Table/14-Max}] initiating a split of this range at key /Table/15 [r12]
I180816 05:21:31.793567 1831 sql/event_log.go:126  [n1,intExec=initializeClusterSecret] Event: "set_cluster_setting", target: 0, info: {SettingName:cluster.secret Value:gen_random_uuid()::STRING User:root}
I180816 05:21:31.880250 1835 storage/replica_command.go:282  [split,n1,s1,r12/1:/{Table/15-Max}] initiating a split of this range at key /Table/16 [r13]
I180816 05:21:31.910793 1735 sql/event_log.go:126  [n1,intExec=create-default-db] Event: "create_database", target: 50, info: {DatabaseName:defaultdb Statement:CREATE DATABASE IF NOT EXISTS defaultdb User:root}
I180816 05:21:31.982319 1008 sql/event_log.go:126  [n1,intExec=create-default-db] Event: "create_database", target: 51, info: {DatabaseName:postgres Statement:CREATE DATABASE IF NOT EXISTS postgres User:root}
I180816 05:21:32.026905 1839 storage/replica_command.go:282  [split,n1,s1,r13/1:/{Table/16-Max}] initiating a split of this range at key /Table/17 [r14]
I180816 05:21:32.057074 1420 server/server.go:1620  [n1] done ensuring all necessary migrations have run
I180816 05:21:32.057178 1420 server/server.go:1623  [n1] serving sql connections
I180816 05:21:32.082540 1860 server/server_update.go:67  [n1] no need to upgrade, cluster already at the newest version
I180816 05:21:32.134154 1862 sql/event_log.go:126  [n1] Event: "node_join", target: 1, info: {Descriptor:{NodeID:1 Address:{NetworkField:tcp AddressField:127.0.0.1:35017} Attrs: Locality: ServerVersion:2.0-11 BuildTag:v2.1.0-alpha.20180702-1631-g9568892 StartedAt:1534396889956809155 LocalityAddress:[]} ClusterID:c04613dd-addc-439f-8f56-73e6767bb7e4 StartedAt:1534396889956809155 LastUp:1534396889956809155}
I180816 05:21:32.137470 1877 storage/replica_command.go:282  [split,n1,s1,r14/1:/{Table/17-Max}] initiating a split of this range at key /Table/18 [r15]
I180816 05:21:32.206801 1881 storage/replica_command.go:282  [split,n1,s1,r15/1:/{Table/18-Max}] initiating a split of this range at key /Table/19 [r16]
I180816 05:21:32.297629 1910 storage/replica_command.go:282  [split,n1,s1,r16/1:/{Table/19-Max}] initiating a split of this range at key /Table/20 [r17]
I180816 05:21:32.438091 1954 storage/replica_command.go:282  [split,n1,s1,r17/1:/{Table/20-Max}] initiating a split of this range at key /Table/21 [r18]
I180816 05:21:32.493263 1958 storage/replica_command.go:282  [split,n1,s1,r18/1:/{Table/21-Max}] initiating a split of this range at key /Table/22 [r19]
I180816 05:21:32.591145 1972 storage/replica_command.go:282  [split,n1,s1,r19/1:/{Table/22-Max}] initiating a split of this range at key /Table/23 [r20]
I180816 05:21:32.647608 2005 storage/replica_command.go:282  [split,n1,s1,r20/1:/{Table/23-Max}] initiating a split of this range at key /Table/50 [r21]
I180816 05:21:32.694451 1968 storage/replica_command.go:282  [split,n1,s1,r21/1:/{Table/50-Max}] initiating a split of this range at key /Table/51 [r22]
I180816 05:21:32.933165 1870 sql/event_log.go:126  [n1,client=127.0.0.1:53860,user=root] Event: "set_cluster_setting", target: 0, info: {SettingName:changefeed.experimental_poll_interval Value:0s User:root}
I180816 05:21:32.965498 1870 sql/event_log.go:126  [n1,client=127.0.0.1:53860,user=root] Event: "create_database", target: 52, info: {DatabaseName:d Statement:CREATE DATABASE d User:root}
I180816 05:21:32.967125 2011 storage/replica_command.go:282  [split,n1,s1,r22/1:/{Table/51-Max}] initiating a split of this range at key /Table/52 [r23]
I180816 05:21:33.064235 1854 storage/replica_command.go:282  [split,n1,s1,r23/1:/{Table/52-Max}] initiating a split of this range at key /Table/53 [r24]
I180816 05:21:33.075274 1870 sql/event_log.go:126  [n1,client=127.0.0.1:53860,user=root] Event: "create_table", target: 53, info: {TableName:d.public.foo Statement:CREATE TABLE foo (a INT PRIMARY KEY, b STRING) User:root}
I180816 05:21:33.200314 2050 storage/replica_command.go:282  [split,n1,s1,r24/1:/{Table/53-Max}] initiating a split of this range at key /Table/54 [r25]
I180816 05:21:33.212839 1870 sql/event_log.go:126  [n1,client=127.0.0.1:53860,user=root] Event: "create_table", target: 54, info: {TableName:d.public.bar Statement:CREATE TABLE bar (a INT PRIMARY KEY, b STRING) User:root}
==================
WARNING: DATA RACE
Write at 0x00c4216567b0 by goroutine 53:
  github.com/cockroachdb/cockroach/pkg/roachpb.(*Spans).Swap()
      /go/src/github.com/cockroachdb/cockroach/pkg/roachpb/data.go:1524 +0xcf
  sort.insertionSort()
      /usr/local/go/src/sort/sort.go:28 +0x64
  sort.quickSort()
      /usr/local/go/src/sort/sort.go:209 +0x219
  sort.Sort()
      /usr/local/go/src/sort/sort.go:218 +0x8a
  github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.equalSpanSets()
      /go/src/github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/poller.go:112 +0xb5
  github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.(*poller).Run()
      /go/src/github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/poller.go:153 +0x1433
  github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.(*poller).Run-fm()
      /go/src/github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeed.go:125 +0x55
  github.com/cockroachdb/cockroach/pkg/util/ctxgroup.Group.GoCtx.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:184 +0x6e
  github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup.(*Group).Go.func1()
      /go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup/errgroup.go:58 +0x68

Previous read at 0x00c4216567b0 by goroutine 118:
  github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.makeSpanFrontier()
      /go/src/github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/span_frontier.go:108 +0x19e
  github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.runChangefeedFlow()
      /go/src/github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeed.go:127 +0x746
  github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.changefeedPlanHook.func2()
      /go/src/github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeed_stmt.go:189 +0x12be
  github.com/cockroachdb/cockroach/pkg/sql.(*hookFnNode).startExec.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/planhook.go:140 +0xcf

Goroutine 53 (running) created at:
  github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup.(*Group).Go()
      /go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup/errgroup.go:55 +0x73
  github.com/cockroachdb/cockroach/pkg/util/ctxgroup.Group.GoCtx()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:183 +0xf1
  github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.runChangefeedFlow()
      /go/src/github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeed.go:125 +0x721
  github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.changefeedPlanHook.func2()
      /go/src/github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeed_stmt.go:189 +0x12be
  github.com/cockroachdb/cockroach/pkg/sql.(*hookFnNode).startExec.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/planhook.go:140 +0xcf

Goroutine 118 (running) created at:
  github.com/cockroachdb/cockroach/pkg/sql.(*hookFnNode).startExec()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/planhook.go:139 +0x248
  github.com/cockroachdb/cockroach/pkg/sql.startExec.func2()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/plan.go:568 +0xd3
  github.com/cockroachdb/cockroach/pkg/sql.(*planVisitor).visitInternal.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/walk.go:127 +0x99
  github.com/cockroachdb/cockroach/pkg/sql.(*planVisitor).visitInternal()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/walk.go:552 +0x583
  github.com/cockroachdb/cockroach/pkg/sql.(*planVisitor).visit()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/walk.go:94 +0xfb
  github.com/cockroachdb/cockroach/pkg/sql.walkPlan()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/walk.go:58 +0x1c1
  github.com/cockroachdb/cockroach/pkg/sql.startExec()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/plan.go:573 +0x152
  github.com/cockroachdb/cockroach/pkg/sql.(*planNodeToRowSource).Next()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/plan_node_to_row_source.go:146 +0xbad
  github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.Run()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/base.go:170 +0x42
  github.com/cockroachdb/cockroach/pkg/sql.(*planNodeToRowSource).Run()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/plan_node_to_row_source.go:250 +0xe3
  github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*Flow).StartSync()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/flow.go:590 +0x329
  github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).Run()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:235 +0xb44
  github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).PlanAndRun()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:583 +0x4d6
  github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execWithDistSQLEngine()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:952 +0x28e
  github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).dispatchToExecutionEngine()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:818 +0x911
  github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInOpenState()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:401 +0x108d
  github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:95 +0x46e
  github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).run()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1078 +0x3413
  github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:416 +0x1e0
  github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).serveImpl.func3()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:267 +0x1a9
==================
==================
WARNING: DATA RACE
Write at 0x00c421656780 by goroutine 53:
  github.com/cockroachdb/cockroach/pkg/roachpb.(*Spans).Swap()
      /go/src/github.com/cockroachdb/cockroach/pkg/roachpb/data.go:1524 +0x14e
  sort.insertionSort()
      /usr/local/go/src/sort/sort.go:28 +0x64
  sort.quickSort()
      /usr/local/go/src/sort/sort.go:209 +0x219
  sort.Sort()
      /usr/local/go/src/sort/sort.go:218 +0x8a
  github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.equalSpanSets()
      /go/src/github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/poller.go:112 +0xb5
  github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.(*poller).Run()
      /go/src/github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/poller.go:153 +0x1433
  github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.(*poller).Run-fm()
      /go/src/github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeed.go:125 +0x55
  github.com/cockroachdb/cockroach/pkg/util/ctxgroup.Group.GoCtx.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:184 +0x6e
  github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup.(*Group).Go.func1()
      /go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup/errgroup.go:58 +0x68

Previous read at 0x00c421656780 by goroutine 118:
  github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.makeSpanFrontier()
      /go/src/github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/span_frontier.go:108 +0x19e
  github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.runChangefeedFlow()
      /go/src/github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeed.go:127 +0x746
  github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.changefeedPlanHook.func2()
      /go/src/github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeed_stmt.go:189 +0x12be
  github.com/cockroachdb/cockroach/pkg/sql.(*hookFnNode).startExec.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/planhook.go:140 +0xcf

Goroutine 53 (running) created at:
  github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup.(*Group).Go()
      /go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup/errgroup.go:55 +0x73
  github.com/cockroachdb/cockroach/pkg/util/ctxgroup.Group.GoCtx()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:183 +0xf1
  github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.runChangefeedFlow()
      /go/src/github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeed.go:125 +0x721
  github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.changefeedPlanHook.func2()
      /go/src/github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeed_stmt.go:189 +0x12be
  github.com/cockroachdb/cockroach/pkg/sql.(*hookFnNode).startExec.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/planhook.go:140 +0xcf

Goroutine 118 (running) created at:
  github.com/cockroachdb/cockroach/pkg/sql.(*hookFnNode).startExec()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/planhook.go:139 +0x248
  github.com/cockroachdb/cockroach/pkg/sql.startExec.func2()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/plan.go:568 +0xd3
  github.com/cockroachdb/cockroach/pkg/sql.(*planVisitor).visitInternal.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/walk.go:127 +0x99
  github.com/cockroachdb/cockroach/pkg/sql.(*planVisitor).visitInternal()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/walk.go:552 +0x583
  github.com/cockroachdb/cockroach/pkg/sql.(*planVisitor).visit()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/walk.go:94 +0xfb
  github.com/cockroachdb/cockroach/pkg/sql.walkPlan()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/walk.go:58 +0x1c1
  github.com/cockroachdb/cockroach/pkg/sql.startExec()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/plan.go:573 +0x152
  github.com/cockroachdb/cockroach/pkg/sql.(*planNodeToRowSource).Next()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/plan_node_to_row_source.go:146 +0xbad
  github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.Run()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/base.go:170 +0x42
  github.com/cockroachdb/cockroach/pkg/sql.(*planNodeToRowSource).Run()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/plan_node_to_row_source.go:250 +0xe3
  github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*Flow).StartSync()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/flow.go:590 +0x329
  github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).Run()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:235 +0xb44
  github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).PlanAndRun()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:583 +0x4d6
  github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execWithDistSQLEngine()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:952 +0x28e
  github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).dispatchToExecutionEngine()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:818 +0x911
  github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInOpenState()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:401 +0x108d
  github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:95 +0x46e
  github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).run()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1078 +0x3413
  github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:416 +0x1e0
  github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).serveImpl.func3()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:267 +0x1a9
==================
I180816 05:21:33.526809 1420 util/stop/stopper.go:537  quiescing; tasks left:
1      [async] transport racer
I180816 05:21:33.695235 1666 kv/transport_race.go:67  transport race promotion: ran 52 iterations on up to 864 requests
--- FAIL: TestChangefeedMultiTable (4.06s)
	testing.go:730: race detected during execution of test

```

--- a/pkg/ccl/changefeedccl/poller.go
+++ b/pkg/ccl/changefeedccl/poller.go
@@ -109,6 +109,8 @@ func equalSpanSets(a, b roachpb.Spans) bool {
 	if len(a) != len(b) {
 		return false
 	}
+	a = append(roachpb.Spans(nil), a...)
+	b = append(roachpb.Spans(nil), b...)
 	sort.Sort(a)
 	sort.Sort(b)
 	for i := range a {

